helm/nats-operator/templates/deployment.yaml

```diff
@@ -0,0 +1,114 @@
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: {{ template "nats.fullname" . }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: "{{ template "nats.name" . }}"
+      chart: "{{ template "nats.chart" . }}"
+      release: "{{ .Release.Name }}"
+  template:
+    metadata:
+      labels:
+        app: "{{ template "nats.name" . }}"
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
+      {{- if .Values.podLabels }}
+        {{ .Values.podLabels}}
+      {{- end }}
+      {{- if .Values.podAnnotations }}
+      annotations:
+        {{ .Values.podAnnotations }}
+      {{- end }}
+    spec:
+      updateStrategy:
+        type: {{ .Values.updateStrategy }}
+        {{- if .Values.rollingUpdatePartition }}
+        rollingUpdate:
+          partition: {{ .Values.rollingUpdatePartition }}
+        {{- end }}
+      {{- if .Values.rbacEnabled }}
+      serviceAccountName: nats-operator
+      {{- end }}
+      containers:
+      - name: nats-operator
+        image: {{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}      
+        imagePullPolicy: {{ .Values.pullPolicy  }}
+        env:
+        - name: MY_POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: MY_POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+          ports:
+          - name: readyz 
+            port: 8080
```

Comment:
I think this one has to be `containerPort`? https://github.com/nats-io/nats-operator/blob/68079dc2093aba61711a8a19ae0989bd181b9cf4/example/deployment-rbac.yaml#L29-L35

```
        ports:
        - name: readyz
          containerPort: 8080
```